src/roswire/definitions/package.py

```diff
@@ -90,7 +91,25 @@ def paths(shell: ShellProxy) -> List[str]:
         code, path_str, duration = shell.execute('echo "${ROS_PACKAGE_PATH}"')
         if code != 0:
             raise Exception("unexpected error when fetching ROS_PACKAGE_PATH")
-        paths: List[str] = path_str.strip().split(':')
+        package_paths: List[str] = path_str.strip().split(':')
+        paths: List[str] = []
+        for path in package_paths:
+            cmd = f'test -f {shlex.quote(path)}/package.xml'
```

Comment:
This is better, but it makes more sense to add `FileProxy` to the args for this function and to use its interface to perform this operation.